src/controllers/product.js

```diff
@@ -235,11 +235,48 @@ const orderRequest = async (req, res) => {
   }
 };
 
+const getRequesters = async (req, res) => {
+  try {
+    const product = await ProductModel.findById(req.params.productId);
+    if (!product) {
+      return res.status(404).json({
+        message: 'Product not found',
+      });
+    }
+
+    if (String(product.publisher) !== req.user.userId) {
+      return res.status(401).json({
+        message: 'You are not authorized to perform this action',
+      });
+    }
+
+    if (product.postType !== 'Donate') {
+      return res.status(400).json({
+        message: 'This is not a donation product: no requesters',
+      });
+    }
+
+    const requesters = await ProductModel.findById(
+      req.params.productId
+    ).populate('orderRequests', 'firstName lastName username email');
+
+    if (requesters.orderRequests.length === 0) {
+      return res.status(400).json({
+        message: 'No requesters found for this product',
+      });
+    }
+    return res.status(200).json(requesters);
```

Comment:
we can create a response constant before return like below and then send the response in json so we dont need to show all info of the product
const response = {
      ID: product._id,
      Title: product.title,
      Requesters: product.orderRequests,
    };